query_exp.go

```diff
@@ -0,0 +1,203 @@
+	#include <qdb/client.h>
+
+	qdb_int_t get_int64_from_payload(qdb_point_result_t result)
+	{
+		return (qdb_int_t)result.payload.int64_.value;
+	}
+
+	double get_double_from_payload(qdb_point_result_t result)
+	{
+		return (double)result.payload.double_.value;
+	}
+
+	void get_blob_from_payload(qdb_point_result_t result, const void ** content, qdb_size_t *length)
+	{
+		*content = result.payload.blob.content;
+		*length = result.payload.blob.content_length;
+	}
+
+
+	qdb_timespec_t get_timestamp_from_payload(qdb_point_result_t result)
+	{
+		return (qdb_timespec_t)result.payload.timestamp.value;
+	}
+*/
+import "C"
+import (
+	"time"
+	"unsafe"
+)
+
+// QueryResultValueType : an enum of possible query point result types
+type QueryResultValueType int64
+
+// QueryResultNone : query result value none
+// QueryResultDouble : query result value double
+// QueryResultBlob : query result value blob
+// QueryResultInt64 : query result value int64
+// QueryResultTimestamp : query result value timestamp
+const (
+	QueryResultNone      QueryResultValueType = C.qdb_query_result_none
+	QueryResultDouble    QueryResultValueType = C.qdb_query_result_double
+	QueryResultBlob      QueryResultValueType = C.qdb_query_result_blob
+	QueryResultInt64     QueryResultValueType = C.qdb_query_result_int64
+	QueryResultTimestamp QueryResultValueType = C.qdb_query_result_timestamp
+)
+
+// QueryPointResult : a query result point
+type QueryPointResult struct {
+	valueType QueryResultValueType
+	value     interface{}
+}
+
+// Type : gives the type of the query point result
+func (r QueryPointResult) Type() QueryResultValueType {
+	return r.valueType
+}
+
+// Value : gives the interface{} value of the query point result
+func (r QueryPointResult) Value() interface{} {
+	return r.value
+}
+
+// Get : retrieve the raw interface
+func (r C.qdb_point_result_t) Get() QueryPointResult {
+	output := QueryPointResult{valueType: QueryResultValueType(r._type)}
+
+	switch valueType := r._type; valueType {
+	case C.qdb_query_result_double:
+		output.value = float64(C.get_double_from_payload(r))
+	case C.qdb_query_result_blob:
+		var content unsafe.Pointer
+		var contentLength C.qdb_size_t
+		C.get_blob_from_payload(r, &content, &contentLength)
+		output.value = C.GoBytes(content, C.int(contentLength))
+	case C.qdb_query_result_int64:
+		output.value = int64(C.get_int64_from_payload(r))
+	case C.qdb_query_result_timestamp:
+		output.value = C.get_timestamp_from_payload(r).toStructG()
+	}
+	return output
+}
+
+// GetDouble : retrieve a double from the interface
+func (r C.qdb_point_result_t) GetDouble() (float64, error) {
+	if r._type == C.qdb_query_result_double {
+		return float64(C.get_double_from_payload(r)), nil
+	}
+	return 0, makeErrorOrNil(C.qdb_e_operation_not_permitted)
+}
+
+// GetBlob : retrieve a double from the interface
+func (r C.qdb_point_result_t) GetBlob() ([]byte, error) {
+	if r._type == C.qdb_query_result_blob {
+		var content unsafe.Pointer
```

Comment:
Can you refactor the inside of this if statement and the `case C.qdb_query_result_blob` from `Get()` to another function to avoid code duplication, e.g. `getBlobUnsafe()`.